core/selection.js

```diff
@@ -33,9 +34,14 @@ class Selection {
         setTimeout(this.update.bind(this, Emitter.sources.USER), 1);
       }
     });
-    this.emitter.on(Emitter.events.SCROLL_BEFORE_UPDATE, () => {
+    this.emitter.on(Emitter.events.SCROLL_BEFORE_UPDATE, async (_, mutations) => {
       if (!this.hasFocus()) return;
       const native = this.getNativeRange();
+      // when might need to hack the offset on Safari, when we are dealing with the first character of a row
+      const hackOffset = (native.start.offset === 0 &&
```

Comment:
shouldn't we fix the offset in the `getNativeRange()` function instead where we're already "fixing" it ?

Reply:
I though about that, but then getNativeRange is also used by getRange, setRange and format, which are then used in other places, so it has a greater potential to introducing side effects...